src/main/scala/chisel3/util/experimental/BoringUtils.scala

```diff
@@ -224,9 +224,14 @@ object BoringUtils {
     else if (DataMirror.hasOuterFlip(source)) Flipped(chiselTypeOf(source))
     else chiselTypeOf(source)
     def purePortType = createProbe match {
-      case Some(pi) if pi.writable => RWProbe(purePortTypeBase)
-      case Some(pi)                => Probe(purePortTypeBase)
-      case None                    => purePortTypeBase
+      case Some(pi) =>
+        // If the source is already a probe, don't double wrap it in a probe.
+        purePortTypeBase.probeInfo match {
+          case Some(_)             => purePortTypeBase
+          case None if pi.writable => RWProbe(purePortTypeBase)
+          case None                => Probe(purePortTypeBase)
+        }
+      case None => purePortTypeBase
```

Comment:
Some of the below logic maybe can/should be tweaked, not sure and maybe better to defer that to a later simplification.
For example, if the target is a rwprobe port (new L251, old L246), and this is used by rwTap, it'll add another RWProbe for that.  Possibly error but haven't traced all the paths here...